tests/test_bookmarks_updated.py

```diff
@@ -95,10 +95,21 @@ def bookmarks_test(self, conn_id, testable_streams):
                 # information required for assertions from sync 1 and 2 based on expected values
                 first_sync_count = first_sync_record_count.get(stream, 0)
                 second_sync_count = second_sync_record_count.get(stream, 0)
-                first_sync_messages = [record.get('data') for record in first_sync_records.get(stream, {}).get('messages', [])
-                                       if record.get('action') == 'upsert']
-                second_sync_messages = [record.get('data') for record in second_sync_records.get(stream, {}).get('messages', [])
-                                        if record.get('action') == 'upsert']
+
+                if stream != 'metafields':
+                    first_sync_messages = [record.get('data') for record in first_sync_records.get(stream, {}).get('messages', [])
+                                           if record.get('action') == 'upsert']
+                else:
+                    first_sync_messages = [record.get('data') for record in first_sync_records.get(stream, {}).get('messages', [])
+                                           if record.get('action') == 'upsert' and record.get('data').get('owner_resource') == 'shop']
+
+                if stream != 'metafields':
+                    second_sync_messages = [record.get('data') for record in second_sync_records.get(stream, {}).get('messages', [])
+                                            if record.get('action') == 'upsert']
+                else:
+                    second_sync_messages = [record.get('data') for record in second_sync_records.get(stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert' and record.get('data').get('owner_resource') == 'shop']
```

Comment:
Add a comment here regarding the specific behavior of metafields stream.

Reply:
added comments